src/lib.rs

```diff
@@ -15,8 +16,9 @@
 #![deny(clippy::clone_on_ref_ptr)]
 #![deny(clippy::trivially_copy_pass_by_ref)]
 #![deny(clippy::disallowed_methods)] // [nursery] mark some functions as verboten
-#![deny(clippy::missing_const_for_fn)] // [nursery] valuable to the optimizer,
-                                       // but may produce false positives
+#![deny(clippy::missing_const_for_fn)] // [nursery] valuable to the optimizer, but may produce false positives
+
+//! This is the main entry point for Vector.
```

Comment:
That's a little misleading IMO. The main entry point would be `main.rs` I would think. This would be where we say it is the main library to support building Vector.